mixer/pkg/api/grpcServer.go

```diff
@@ -87,6 +91,34 @@ func (s *grpcServer) Check(legacyCtx legacyContext.Context, req *mixerpb.CheckRe
 	protoBag := attribute.NewProtoBag(&req.Attributes, s.globalDict, s.globalWordList)
 	defer protoBag.Done()
 
+	var resp *mixerpb.CheckResponse
```

Comment:
not a blocker:  do you think it would be valuable to split this method up a bit now?  something like:

```
resp := valueFromCache(s.cache, protoBag)
if resp == nil {
   resp, err = check(legacyCtx, protoBag)
   s.cache.Set(resp, protoBag)
   ...
}
if !allocateQuota(resp) {
   return resp, nil
}
err := quota(resp)
return resp, err
```